### PR TITLE
Protect: Add new Debug Helper module and fix a bug

### PR DESCRIPTION
**Pull request by leogermani**
This PR adds a new module to the Jetpack Debugger plugin that allows you to set different statuses for the Protect plugin. This is very handy when developing the plugin, since you need a way to simulate the different responses the API might return.

While working on this, I discovered a small bug in the plugin, that would keep Core vuls from showing up.

#### Changes proposed in this Pull Request:

* Create new Protect Helper debug module
* Fix a bug in Protect plugin keeping the core vulnerabilities from showing up

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Go to Jetpack Debug and activate the "Protect Helper" module
* Go to Jetpack Debug > Protect helper
* See if you can understand what this screen does
* Play with the options there
* Visit Jetpack Protect page and make sure the page content follows what you are configuring in the Protect Helper plugin

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202476778482733

### CODE REVIEW

**Comment by singerb** [comment]
@nateweller tagging you here since your PR to migrate the data format handling may (or may not)  be impacted by the same bug.

**Comment by nateweller** [comment]
> @nateweller tagging you here since your PR to migrate the data format handling may (or may not) be impacted by the same bug.

Thanks @singerb, I'll make sure #24761 is updated - looks like it might have the same issue.

File: projects/plugins/debug-helper/modules/class-protect-helper.php [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+		<form action="<?php echo esc_url( admin_url( 'admin-post.php' ) ); ?>" method="post">
+
+		<table class="form-table" role="presentation">
+		<tbody>
+		<tr>
+			<th scope="row">Overwrite Protect Status</th>
+			<td>
+				<fieldset>
+					<p>If enable, Protet status fetched from the server will be ignored and the rules below will define the current status.</p>
```
Comment (nateweller) [inline projects/plugins/debug-helper/modules/class-protect-helper.php]:
```suggestion
					<p>If enabled, the Protect status fetched from the server will be ignored and the rules below will define the current status.</p>
```
Minor typos

File: projects/plugins/debug-helper/modules/class-protect-helper.php [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+					<option value="error" <?php selected( 'error', $settings['status'] ); ?>>Error - When there's an error fetching the status</option>
+					<option value="incomplete" <?php selected( 'incomplete', $settings['status'] ); ?>>Incomplete - When we have results, but some extensions were not scanned</option>
+					<option value="complete" <?php selected( 'complete', $settings['status'] ); ?>>Complete - When we have results for all extensions</option>
+				</select>
+			</td>
+		</tr>
+		<tr>
+			<th scope="row">Vulnerabilities</th>
+			<td>
+				<label><input type="checkbox" name="vuls_for_core" <?php echo ( $settings['vuls_for_core'] ? 'checked="checked"' : '' ); ?>> Add vulnerabilities to core</label><br>
```
Comment (nateweller) [inline projects/plugins/debug-helper/modules/class-protect-helper.php]:
```suggestion
				<label><input type="checkbox" name="vuls_for_core" <?php echo ( $settings['vuls_for_core'] ? 'checked="checked"' : '' ); ?>> Add vulnerabilities to Core</label><br>
```
Very minor suggestion - capitalize "Core" to match "Themes" and "Plugins" in the same list.

File: projects/plugins/debug-helper/modules/class-protect-helper.php [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+	 */
+	public function render_ui() {
+		$settings = $this->get_stored_settings();
+		?>
+		<h1>Protect Helper</h1>
+
+		<form action="<?php echo esc_url( admin_url( 'admin-post.php' ) ); ?>" method="post">
+
+		<table class="form-table" role="presentation">
+		<tbody>
```
Comment (nateweller) [inline projects/plugins/debug-helper/modules/class-protect-helper.php]: `tbody` and content needs one more level of indentation

**Review by nateweller — DISMISSED** [review]
Left a few minor notes, but this otherwise looks perfect. This helper is going to be fantastic for testing and development, thanks so much for this work!

File: projects/plugins/debug-helper/modules/class-protect-helper.php (hunk 3 above)
Comment (leogermani) [reply projects/plugins/debug-helper/modules/class-protect-helper.php]: fixed

**Review by retrofox — APPROVED** [review]
LGTM

**Pull request merged**